pkgs/by-name/me/melonDS/package.nix

```
@@ -46,16 +48,26 @@ stdenv.mkDerivation (finalAttrs: {
     libGL
     qtbase
     qtmultimedia
-    wayland
     zstd
+  ] ++ lib.optionals stdenv.isLinux [
+    wayland
   ];
 
   strictDeps = true;
 
-  qtWrapperArgs = [
+  qtWrapperArgs = lib.optionals stdenv.isLinux [
     "--prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libpcap ]}"
+  ] ++ lib.optionals stdenv.isDarwin [
+    "--prefix DYLD_LIBRARY_PATH : ${lib.makeLibraryPath [ libpcap ]}"
   ];
 
+  installPhase = lib.optionalString stdenv.isDarwin ''
+    runHook preInstall
+    mkdir -p $out/Applications
+    cp -r melonDS.app $out/Applications/
```

```suggestion
    mv melonDS.app $out/Applications/
```
this will help save a bit of space when building

---

Does it make any difference? Garbage-collect and let it go.

---

Does it make any difference? Garbage-collect and let it go.
This is not as if we are dealing with embedded DS...

---

> Does it make any difference?

yes; it should be faster and not cause duplication of what's basically all of $out. 

> This is not as if we are dealing with embedded DS...

does that mean we should just slow down builds and take up space for no reason? i don't understand your point here. there isn't any reason `cp` should be preferred here

---

> yes; it should be faster and not cause duplication of what's basically all of $out.

Why? There are no guarantees that the files are in the same filesystem, as far as I remember.

Also, moving things around - copy-then-delete - is not how the typical installation scripts do the things. 
They usually employ the `install` command - that has [some differences from `cp`](https://unix.stackexchange.com/questions/104982/why-use-install-rather-than-cp-and-mkdir) - instead of `mv`.

And, taking out space?
This is not an issue - or else each and every package manager should run `make distclean`  or `rm -fr $builderDirectory` as its last phase.